### PR TITLE
 modify commitBlock to in case of lock-contention caused unordered block commit

### PR DESCRIPTION
**Pull request by cyjseagull**

### CODE REVIEW

File: libconsensus/pbft/PBFTEngine.h [hunk 1]
```diff
@@ -346,6 +347,15 @@ class PBFTEngine : public ConsensusEngineBase
     template <class T>
     inline CheckResult checkReq(T const& req, std::ostringstream& oss) const
     {
+        /// TODO: add this logic when block sync is faster
```
Comment (wheatli) [inline libconsensus/pbft/PBFTEngine.h]: just delete them if they are useless

File: libconsensus/pbft/PBFTEngine.h [hunk 2]
```diff
@@ -404,6 +414,17 @@ class PBFTEngine : public ConsensusEngineBase
         }
         return false;
     }
+
+    /// TODO: add this logic when block sync is faster
```
Comment (wheatli) [inline libconsensus/pbft/PBFTEngine.h]: the same as before

Comment (cyjseagull) [reply libconsensus/pbft/PBFTEngine.h]: done.

File: libconsensus/pbft/PBFTEngine.h (hunk 1 above)
Comment (cyjseagull) [reply libconsensus/pbft/PBFTEngine.h]: done

File: libconsensus/pbft/PBFTEngine.h [hunk 3]
```diff
@@ -484,6 +490,19 @@ class PBFTEngine : public ConsensusEngineBase
         return boost::filesystem::space(path).available > 1024;
     }
 
+    void updateViewMap(IDXTYPE const& idx, VIEWTYPE const& view)
+    {
+        WriteGuard l(x_viewMap);
+        if (idx != nodeIdx())
```
Comment (wheatli) [inline libconsensus/pbft/PBFTEngine.h]: Why do it need to check the value of idx? I think it is ok to just set m_viewMap[idx] as view

Comment (cyjseagull) [reply libconsensus/pbft/PBFTEngine.h]: done.

**Pull request merged**